### PR TITLE
Adding a Swagger view

### PR DESCRIPTION
**Pull request by tarekziade**
- [x] adding Swagger Spec
- [x] tests

### CODE REVIEW

**Comment by tarekziade** [comment]
@leplatrem r?

File: absearch/templates/swagger.tpl [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        description: Performs an heartbeat on the app
+        produces:
+        - application/json
+        responses:
+          '200':
+            description: The hearbeat was successfull
+    /1/{prod}/{ver}/{channel}/{locale}/{territory}/{dist}/{distver}:
+      get:
+        operationId: addUserToCohort
+        description: Assign the user to a cohort and returns its settings
```
Comment (leplatrem) [inline absearch/templates/swagger.tpl]: nit: here you have a capitalized verb, where elsewhere you have a lowercase 3rd person form

File: absearch/templates/swagger.tpl [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        produces:
+        - application/json
+        responses:
+          '200':
+            description: The cohort settings are returns
```
Comment (leplatrem) [inline absearch/templates/swagger.tpl]: typo

File: absearch/templates/swagger.tpl [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+          type: string
+    /1/{prod}/{ver}/{channel}/{locale}/{territory}/{dist}/{distver}/{cohort}:
+      get:
+        operationId: returnCohortSettings
+        description: Returns the cohort settings
+        produces:
+        - application/json
+        responses:
+          '200':
+            description: The hearbeat was successfull
```
Comment (leplatrem) [inline absearch/templates/swagger.tpl]: leftover

File: absearch/tests/test_server.py [hunk 4]
```diff
@@ -30,6 +32,29 @@ def test_info():
...
+
+    assert spec['info']['version'] == __version__
+    assert spec['schemes'] == ['https']
+    assert spec['host'] == 'localhost:80'
+
+    # now testing that every GET endpoint is present
+    for path, items in spec['paths'].items():
+        for verb, options in items.items():
+            verb = verb.upper()
+            if verb != 'GET':
```
Comment (leplatrem) [inline absearch/tests/test_server.py]: nit: can be done in one line

File: absearch/tests/test_server.py [hunk 5]
```diff
@@ -30,6 +32,29 @@ def test_info():
...
+                continue
+            statuses = [int(st) for st in options['responses'].keys()]
+            res = app.get(path, status=statuses)
```
Comment (leplatrem) [inline absearch/tests/test_server.py]: res is not used ?

File: absearch/tests/test_server.py [hunk 6]
```diff
@@ -30,6 +32,29 @@ def test_info():
...
+    assert spec['schemes'] == ['https']
+    assert spec['host'] == 'localhost:80'
+
+    # now testing that every GET endpoint is present
+    for path, items in spec['paths'].items():
+        for verb, options in items.items():
+            verb = verb.upper()
+            if verb != 'GET':
+                continue
+            statuses = [int(st) for st in options['responses'].keys()]
```
Comment (leplatrem) [inline absearch/tests/test_server.py]: the test wants to test that every endpoint is present, but in `statuses` you'll have `404`.

**Pull request merged**